docs/ir/ir-breaking-changes.rst

```diff
@@ -115,6 +115,45 @@ This causes differences in some contracts, for example:
 Previously `f()` would return `0x6465616462656566313564656164000000000000000000000000000000000010` (it has correct length, and correct first 8 elements, but than it contains dirty data which was set via assembly).
 Now it is returning `0x6465616462656566000000000000000000000000000000000000000000000010` (it has correct length, and correct elements, but doesn't contain dirty data).
 
+ * Intermediate values with dirty bits are handled differently in an inline assembly block.
```

Comment:
I think this can go into "internals", and maybe to describe it a bit more concisely: The old code generator only performs cleanup before an operation where it is important to have dirty bits cleaned. The new code generator performs cleanup after each operation that can result in dirty bits.

Reply:
Updated.